File: src/main/java/dev/stashy/extrasounds/sounds/SoundType.java
package dev.stashy.extrasounds.sounds;

import dev.stashy.extrasounds.Mixers;
import net.minecraft.sound.SoundCategory;

public enum SoundType
{
    PICKUP(1f, Mixers.INVENTORY, "item.pickup"),
    PLACE(0.9f, Mixers.INVENTORY, "item.place"),
    HOTBAR(1f, Mixers.SCROLL, "item.select"),
    EFFECT(1f, Mixers.EFFECTS, "effect"),
    CHAT(1f, Mixers.CHAT, "ui.chat"),
    CHAT_MENTION(1f, Mixers.CHAT_MENTION, "ui.chat"),
    TYPING(1f, Mixers.TYPING, "ui.typing"),
    ACTION(1f, Mixers.ACTION, "item.action");

    public final float pitch;
    public final SoundCategory category;
    public final String prefix;

    SoundType(float pitch, SoundCategory category, String prefix)
    {
        this.pitch = pitch;
        this.category = category;
        this.prefix = prefix;
    }
}

File: src/main/kotlin/dev/stashy/extrasounds/sounds/SoundType.kt
package dev.stashy.extrasounds.sounds

import dev.stashy.extrasounds.Mixers
import net.minecraft.sound.SoundCategory

enum class SoundType(val pitch: Float, val category: SoundCategory, val prefix: String) {
    PICKUP(1f, Mixers.Inventory, "item.pickup"),
    PLACE(0.9f, Mixers.Inventory, "item.place"),
    HOTBAR(1f, Mixers.Scroll, "item.select"),
    EFFECT(1f, Mixers.Effects, "effect"),
    CHAT(1f, Mixers.Chat, "ui.chat"),
    CHAT_MENTION(1f, Mixers.ChatMention, "ui.chat"),
    TYPING(1f, Mixers.Typing, "ui.typing"),
    ACTION(1f, Mixers.Action, "action");
}